### PR TITLE
upgrade scripts

### PR DESCRIPTION
**Pull request by yrong**
## Security Checklist

- [ ] 1. Re-Entrancy
- [ ] 2. Arithmetic Over/Under Flows
- [ ] 3. Unexpected Ether
- [ ] 4. Delegatecall
- [ ] 5. Default Visibilities
- [ ] 6. Entropy Illusion
- [ ] 7. External Contract Referencing
- [ ] 8. Short Address/Parameter Attack (off chain)
- [ ] 9. Unchecked CALL Return Values
- [ ] 10. Race Conditions / Front Running
- [ ] 11. Denial Of Service (DOS)
- [ ] 12. Block Timestamp Manipulation
- [ ] 13. Constructors with Care
- [ ] 14. Uninitialized Storage Pointers
- [ ] 15. Floating Points and Precision
- [ ] 16. Tx.Origin Authentication
- [ ] 17. Address.isContract Re-Entrancy via Constructor

### ⚠️ NOTES ⚠️

Make sure to think about each of these exploits in this PR.

### CODE REVIEW

**Comment by yrong** [comment]
@WalidOfNow 
this pr is a sub task for preparing some upgrade scripts, to demonstrate some use cases of upgrability with our diamond proxy I just add a test for temporarily disable functions, but it's not a full test set for upgrading our pool components. remember last time you also mentioned working on adding more tests, do you have any pr for that?

File: test-suites/pool-upgrade.spec.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  before(async () => {
+    testEnv = await loadFixture(testEnvFixture);
+  });
+  beforeEach(async () => {
+    snapShot = await evmSnapshot();
+  });
+
+  afterEach(async () => {
+    await evmRevert(snapShot);
+  });
```
Comment (alannotnerd) [inline test-suites/pool-upgrade.spec.ts:35]:
```suggestion
```

File: test-suites/pool-upgrade.spec.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
```
Comment (alannotnerd) [inline test-suites/pool-upgrade.spec.ts:36]:
```suggestion
```

File: test-suites/pool-upgrade.spec.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  it("upgrade:disable liquidation by remove liquidationCall in current pool", async () => {
+    const {
+      addressesProvider,
+      weth,
+      usdc,
+      users: [borrower, liquidator],
+    } = testEnv;
```
Comment (alannotnerd) [inline test-suites/pool-upgrade.spec.ts:43]:
```suggestion
    } = await loadFixture(testEnvFixture);
```

File: test-suites/pool-upgrade.spec.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    );
+  });
+
+  it("upgrade:disable liquidation by upgrading to a new pool contract which will revert liquidationCall", async () => {
+    const {
+      addressesProvider,
+      weth,
+      usdc,
+      users: [borrower, liquidator],
+    } = testEnv;
```
Comment (alannotnerd) [inline test-suites/pool-upgrade.spec.ts:92]:
```suggestion
    } = await loadFixture(testEnvFixture);
```

**Review by GopherJ — APPROVED** [review]

**Pull request merged**